app/javascript/packs/project_answers.js

```diff
@@ -122,9 +151,34 @@ export async function main() {
           },
          )
         .then((response) => {
+            // User feedback.
+            input.classList.remove('autosave-input-error');
+            inputAlert.innerHTML = "";
+            setTimeout(function(){
+                if (getUnsavedInputs().length === 0) {
+                    changeAutoSaveStatus('All progress saved.', 'autosave-status-success');
+                } else {
+                    changeAutoSaveStatus('Some answers are still unsaved!', 'autosave-status-error');
+                }
+            }, 400); // Wait a split second to change status back so the 'Saving...' message is actually visible
+
+            // Logging.
             honeySpan.addField('response.status', response.status, false);
         })
         .catch((error) => {
+            // User feedback.
+            input.classList.add('autosave-input-error');
+            inputAlert.innerHTML = "Failed to save answer. <a href='#'>Retry?</a>";
+            inputAlert.querySelector('a').onclick = () => { 
+                // clicking the Retry on any of the failed answers retries them all
+                let allUnsavedInputs = getUnsavedInputs();
+                allUnsavedInputs.forEach((input) => {
+                    sendAnswer({ target: input }, true);
```

Comment:
This felt a little weird faking an event object by just passing a plain ol' javascript object with a target attribute. Does this seem like the right "javascripty" way to do this? I didn't think it was worth adding code inside the sendAnswer function to check the type of what was passed in.

Reply:
Since we already have a reference to `input`, I think this would be better as 
```suggestion
                    input.dispatchEvent(new Event('blur'));
```